Content/Tweaks/RespawnTimeModifier/RespawnTimeModifierPlayer.cs

```diff
@@ -0,0 +1,45 @@
+﻿using AdLibitum.Configuration.Server;
+using System;
+using Terraria;
+using Terraria.ModLoader;
+using Terraria.ID;
+
+namespace AdLibitum.Content.Tweaks.RespawnTimeModifier
+{
+    public class RespawnTimeModifierPlayer : ModPlayer
+    {
+        public override void UpdateDead() {
+            if (AmICloseToRespawnTimeAffectingBoss())
+                return;
+
+            int respawnTimeMod = StandardServerConfig.Config.Tweaks.RespawnTimeModifier;
+
+            // 0 / 100 == 1
+            if (respawnTimeMod == 0)
+                Player.respawnTimer = 0;
+
+            int targetRespawnTime = (int)(MaxRespawnTime() * (respawnTimeMod / 100f));
+
+            if ((Player.respawnTimer > targetRespawnTime && respawnTimeMod > 100) ||
+                (Player.respawnTimer < targetRespawnTime && respawnTimeMod < 100))
+                Player.respawnTimer = targetRespawnTime;
```

Comment:
Could this just be shortened with a clamp?

Reply:
I don't understand what you mean. How could a clamp be used here?

Reply:
Ah, took a closer look at it. A clamp could be used, but it's rather verbose. Fine as is.